src/System.Private.CoreLib/shared/System/Text/EncodingTable.cs

```diff
@@ -0,0 +1,182 @@
+
+        internal static int GetCodePageFromName(string name)
+        {
+            if (name == null)
+                throw new ArgumentNullException(nameof(name));
+
+            int code;
+            lock (s_name2code)
+            {
+                if (s_name2code.TryGetValue(name, out code))
+                    return code;
+            }
+
+            code = InternalGetCodePageFromName(name);
+
+            lock (s_name2code)
+            {
+                if (!s_name2code.TryAdd(name, code))
+                    code = s_name2code[name];
+            }
+
+            return code;
+        }
+
+        // Find the data item by binary searching the table.
+        private static int InternalGetCodePageFromName(string name)
+        {
+            int left = 0;
+            int right = s_encodingNameIndices.Length - 2;
+            int index;
+            int result;
+
+            Debug.Assert(s_encodingNameIndices.Length == s_codePagesByName.Length + 1);
+            Debug.Assert(s_encodingNameIndices[s_encodingNameIndices.Length - 1] == s_encodingNames.Length);
+
+            name = name.ToLowerInvariant();
+
+            //Binary search the array until we have only a couple of elements left and then
+            //just walk those elements.
+            while ((right - left) > 3)
+            {
+                index = ((right - left) / 2) + left;
+
+                Debug.Assert(index < s_encodingNameIndices.Length - 1);
+                result = CompareOrdinal(name, s_encodingNames, s_encodingNameIndices[index], s_encodingNameIndices[index + 1] - s_encodingNameIndices[index]);
+                if (result == 0)
+                {
+                    //We found the item, return the associated codePage.
+                    return s_codePagesByName[index];
+                }
+                else if (result < 0)
+                {
+                    //The name that we're looking for is less than our current index.
+                    right = index;
+                }
+                else
+                {
+                    //The name that we're looking for is greater than our current index
+                    left = index;
+                }
+            }
+
+            //Walk the remaining elements (it'll be 3 or fewer).
+            for (; left <= right; left++)
+            {
+                Debug.Assert(left < s_encodingNameIndices.Length - 1);
+                if (CompareOrdinal(name, s_encodingNames, s_encodingNameIndices[left], s_encodingNameIndices[left + 1] - s_encodingNameIndices[left]) == 0)
+                {
+                    return s_codePagesByName[left];
+                }
+            }
+
+            // The encoding name is not valid.
+            throw new ArgumentException(
+                SR.Format(SR.Argument_EncodingNotSupported, name),
+                nameof(name));
+        }
+
+        private static int CompareOrdinal(string s1, string s2, int index, int length)
```

Comment:
This should be deleted and code changed to use `internal static int CompareOrdinal(ReadOnlySpan<char> strA, ReadOnlySpan<char> strB)` instead.